### PR TITLE
update to yarn 3

### PR DESCRIPTION
**Pull request by Dhaulagiri**
### :pushpin: Summary

This PR moves us from yarn v1 to yarn v3

### :hammer_and_wrench: Detailed description

The main impetus for this that yarn v1 is in maintenance mode and support for workspaces in it is pretty rough. More context is in #341, but in short none of the small bugs we are experiencing in v1 will ever be fixed so this is the best way to get onto a more modern version of yarn.

The one drawback is that this change exposed an underlying issue with our ember-3.24 scenario for the components package. This is a quirk of the monorepo and ember that actually exists locally regardless, this change just exposes it in CI. We've informally agreed that this is not a long term concern though given we expect most teams to be on more modern versions of ember.

### 👀 How to review

👉 Review commit-by-commit

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/config/ember-try.js [hunk 1]
```diff
@@ -5,6 +5,10 @@ const { embroiderSafe, embroiderOptimized } = require('@embroider/test-setup');
 
 module.exports = async function () {
   return {
+    useYarn: true,
+    buildManagerOptions() {
```
Comment (Dhaulagiri) [inline packages/components/config/ember-try.js:9]: add a comment on why this was needed

**Comment by Dhaulagiri** [comment]
Opening this up for review. I'm honestly not 100% convinced this is the best idea long term as my perception is that, in general, yarn as a project is not in the best place (yarn 1 is obsolete but widely used, yarn 3 is more actively worked on but not used much from what I can see, etc). Even if we accept this PR it may be worth exploring how to finalize a migration to use npm which seems somewhat more common 🤷

**Review by alex-ju — APPROVED** [review]
I guess it's a bet on increased adoption of `yarn 3` across the web that I'm personally willing to make. Extra 🍬 for updating the docs!

**Review by MelSumner — APPROVED** [review]
Thank you for doing this work! 👍

**Pull request merged**